### PR TITLE
New rules page filters

### PR DESCRIPTION
**Pull request by jeudy100**
screenshot from 2017-08-07 16-59-46

This is for issue #176 and #154

### CODE REVIEW

**Comment by iphands** [comment]
@jeudy100 Is this still waiting on api?

**Comment by jeudy100** [comment]
@iphands yep it is still waiting to be merged

**Comment by jeudy100** [comment]
image

**Comment by ryelo** [comment]
Is it possible for a person to filter, for example, both medium and high impact?

File: app/js/components/filterComponents/incidentsTriState/incidentsTriState.controller.js [hunk 1]
```diff
@@ -49,7 +57,7 @@ function incidentsTriStateCtrl($location, $rootScope, $scope, Events, IncidentFi
 
     function init () {
         $scope.showIncidents = $location.search()[Events.filters.incident] ?
-            $location.search()[Events.filters.incident] : 'all';
+            $location.search()[Events.filters.incident] : FilterService.getIncidents();
```
Comment (Kinlaw) [inline app/js/components/filterComponents/incidentsTriState/incidentsTriState.controller.js]: since FilterService is already setting incidents to either $location.search()[Events.filters.incident] or 'all' you could shorten this to just $scope.showIncidents = FilterService.getIncidents();

File: app/js/components/filterComponents/ansibleSupportTriState/ansibleSupportTriState.controller.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    function getTag () {
+        let tag = AnsibleSupportFilters[$scope.showAnsibleSupport].tag;
+
+        return tag;
+    }
+
+    function init () {
+        $scope.showAnsibleSupport = $location.search()[Events.filters.ansibleSupport] ?
+            $location.search()[Events.filters.ansibleSupport] :
+            FilterService.getAnsibleSupport();
```
Comment (Kinlaw) [inline app/js/components/filterComponents/ansibleSupportTriState/ansibleSupportTriState.controller.js]: FilterService is already setting ansibleSupport so this can be shortened to just $scope.showAnsibleSupport = FilterService.getAnsibleSupport();

File: app/js/components/filterComponents/impactSelect/impactSelect.directive.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    };
+
+    function getTag () {
+        return $scope.selected.tag;
+    }
+
+    function init() {
+        let option = $location.search()[Events.filters.impact] ?
+            $location.search()[Events.filters.impact] :
+            FilterService.getImpact();
```
Comment (Kinlaw) [inline app/js/components/filterComponents/impactSelect/impactSelect.directive.js]: FilterService is already setting the default value here.

File: app/js/components/filterComponents/likelihoodSelect/likelihoodSelect.directive.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    };
+
+    function getTag () {
+        return $scope.selected.tag;
+    }
+
+    function init() {
+        let option = $location.search()[Events.filters.likelihood] ?
+            $location.search()[Events.filters.likelihood] :
+            FilterService.getLikelihood();
```
Comment (Kinlaw) [inline app/js/components/filterComponents/likelihoodSelect/likelihoodSelect.directive.js]: FilterService is already setting the default value.

File: app/js/components/filterComponents/ruleStatusTriState/ruleStatusTriState.controller.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+    function getTag () {
+        let tag = RuleStatusFilters[$scope.showRuleStatus].tag;
+
+        return tag;
+    }
+
+    function init () {
+        $scope.showRuleStatus = $location.search()[Events.filters.ruleStatus] ?
+            $location.search()[Events.filters.ruleStatus] : FilterService.getRuleStatus();
```
Comment (Kinlaw) [inline app/js/components/filterComponents/ruleStatusTriState/ruleStatusTriState.controller.js]: FilterService is already setting the default value.

File: app/js/components/filterComponents/totalRiskSelect/totalRiskSelect.controller.js [hunk 6]
```diff
@@ -53,18 +59,27 @@ function totalRiskSelectCtrl($location,
             });
 
         $scope.label = $scope.selected.label;
-
+        setSeverity($scope.selected.value);
```
Comment (Kinlaw) [inline app/js/components/filterComponents/totalRiskSelect/totalRiskSelect.controller.js]: setSeverity is being set twice in this function. Also, you are using option.value and $scope.selected.value. Since $scope.selected is being set you can probably get rid of the option param.
